### PR TITLE
DetailsList: Add aria label for the group chevron button

### LINKED ISSUE #8158: Accessibility: DetailsList - Chevron button in list header is not read by screen readers
### Environment Information 
 
- __Package version(s)__: 6.142.0

### Please provide a reproduction of the bug in a codepen:
 
https://developer.microsoft.com/en-us/fabric#/components/detailslist/grouped
 
#### Actual behavior:
 
The first chevron button in the list header in Automated checks is not read by ATs.
The first chevron icon, on the header is just read as "clickable"
 
#### Expected behavior:
 
The first chevron icon, on the header should be mentioned by AT as

#### Priorities and help requested:
 
Are you willing to submit a PR to fix? (Yes)
 
Requested priority: (High)

### PR DESCRIPTION
**Pull request by leddie24**
#### Pull request checklist

- [x] Addresses an existing issue: Fixes #8158 
- [x] Include a change request file using `$ npm run change`

#### Description of changes

- Add aria label for the group chevron button. 
- Adds aria-expanded attribute to the group chevron header button

#### Focus areas to test

(optional)

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/8253)

### CODE REVIEW

**Comment by KevinTCoughlin** [comment]
Unrelated to this PR, but I'm looking forward to the day we properly modularize these components so we can target them via the `as` pattern instead of having to drill hills for individual props :).

**Comment by natalieethell** [comment]
@KevinTCoughlin ah, if we want to clarify that this is for the root group, I can't think of anything better than what you suggested, `ariaLabelForToggleAllGroupsButton` or `ariaLabelForExpandCollapseAllGroupsButton`.

**Comment by KevinTCoughlin** [comment]
Pushing up some changes to remove the need to expose the new prop via `IDetailsListProps`, the usage would then be like this:

```tsx
<DetailsList
  componentRef={this._root}
  items={items}
  groups={groups}
  columns={this._columns}
  ariaLabelForSelectAllCheckbox="Toggle selection for all items"
  ariaLabelForSelectionColumn="Toggle selection"
  onRenderDetailsHeader={(props: IDetailsHeaderProps) => {
    return <DetailsHeader {...props} ariaLabelForToggleAllGroupsButton={'Expand collapse groups'} />;
  }}
  groupProps={{
    showEmptyGroups: true
  }}
  onRenderItemColumn={this._onRenderColumn}
  compact={isCompactMode}
/>
```

**Comment by KevinTCoughlin** [comment]
I'll update the API file shortly... then this should be good-to-go!

File: packages/office-ui-fabric-react/src/components/DetailsList/DetailsHeader.base.tsx [hunk 1]
```diff
@@ -270,7 +271,13 @@ export class DetailsHeaderBase extends BaseComponent<IDetailsHeaderBaseProps, ID
             ]
           : null}
         {groupNestingDepth! > 0 && this.props.collapseAllVisibility === CollapseAllVisibility.visible ? (
-          <div className={classNames.cellIsGroupExpander} onClick={this._onToggleCollapseAll} data-is-focusable={true}>
+          <div
+            className={classNames.cellIsGroupExpander}
+            onClick={this._onToggleCollapseAll}
+            data-is-focusable={true}
+            aria-label={ariaLabelForToggleAllGroupsButton}
+            aria-expanded={!isAllCollapsed}
```
Comment (KevinTCoughlin) [inline packages/office-ui-fabric-react/src/components/DetailsList/DetailsHeader.base.tsx]: Thanks for adding this `aria-expanded` @leddie24, very thorough 👍!

**Pull request merged**